src/main/java/sg/dex/starfish/impl/file/FileAsset.java

```diff
@@ -23,96 +21,106 @@
-
-			String content = Utils.stringFromStream(new FileInputStream(f));
-			String hashOfContent = Hex.toString(Hash.keccak256(content));
-
-			Map<String, Object> ob = Utils.createDefaultMetadata();
-			ob.put(TYPE, DATA_SET);
-			ob.put(SIZE, f.length());
-			ob.put(FILE_NAME, f.getName());
-			ob.put(CONTENT_TYPE, OCTET_STREAM);
-			ob.put(CONTENT_HASH, hashOfContent);
-
-			if (meta != null) {
-				Map<String,Object> metaMap = JSON.toMap(meta);
-				for (Map.Entry<String, Object> me : metaMap.entrySet()) {
-					ob.put(me.getKey(), me.getValue());
-				}
-			}
-
-			return ob;
-
-		} catch (FileNotFoundException e) {
-			throw new StarfishValidationException(e.getMessage());
-		}
-	}
-
-	/**
-	 * Gets an input stream that can be used to consume the content of this asset.
-	 *
-	 * Will throw an exception if consumption of the asset data in not possible
-	 * locally.
-	 * 
-	 * @throws AuthorizationException if requestor does not have access permission
-	 * @throws StorageException if unable to load the Asset
-	 * @return An input stream allowing consumption of the asset data
-	 */
-	@Override
-	public InputStream getContentStream() {
-		try {
-			return new FileInputStream(file);
-		}
-		catch (FileNotFoundException e) {
-			throw new StorageException("File not found ,file : "+file , e);
-		}
-	}
-
-	@Override
-	public long getContentSize() {
-		return null != file ? file.length(): -1;
-	}
-
-
+    private final File file;
+
+    protected FileAsset(String meta, File file) {
+        super(meta);
+        this.file = file;
+    }
+
+
+    /**
+     * This method is to crete File Asset with specific given path.
+     *
+     * @param f path of the file
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f) {
+        return new FileAsset(buildMetadata(f, null), f);
+    }
+
+    /**
+     * This method is to crete Resource Asset with specific given path, metadata
+     *
+     * @param f        path of the file
+     * @param metaData metadata associated with the asset.This metadata will be be added in addition to default
+     *                 metadata i.e DATE_CREATED,TYPE,CONTENT_TYPE.If same key,value is provided then the
+     *                 default value will be overridden.
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f, String metaData) {
+        return new FileAsset(buildMetadata(f, metaData), f);
+    }
+
+    /**
+     * Build default metadata for a file asset
+     *
+     * @param f    The file to use for this file asset
+     * @param meta metadata associated with the asset.This metadata will be be added in addition to default
+     *             metadata i.e DATE_CREATED,TYPE,CONTENT_TYPE.If same key,value is provided then the
+     *             default value will be overridden.
+     * @return The default metadata as a String
+     */
+    protected static String buildMetadata(File f, String meta) {
```

Comment:
Shouldn't have this function signature. If the user want to build default metadata, the argument should be a Map<String,Object>. Only use a String for metadata if you want that *exact* metadata.

Reply:
resolved in latest PR